### DISCUSSION TITLE
Date Input

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#270] [Component suggestions]
<img width="480" alt="visuele weergave van het date input component" src="https://github.com/nl-design-system/backlog/assets/84851545/497a473b-dc61-4910-9575-d1c0e073d7ca">

We willen graag meer leren over het **Date Input** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1718791825868629)

---

## Naam
Date Input

## Doel
Invoerveld om een datum in te vullen met de mogelijkheden die de browser biedt.

## Ook bekend als
Date

## Gerelateerde componenten
[Calendar](https://github.com/orgs/nl-design-system/discussions/273), [Date Input Group](https://github.com/orgs/nl-design-system/discussions/271), [Date Picker](https://github.com/orgs/nl-design-system/discussions/272)

## Onderzoek

[input type ="date" - Mozilla](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date)
Informatie over input type="date", de standaard ingebouwde (native) Date Input.

[Redenen waarom GOV.UK geen gebruikmaakt van native date input - GOV.UK](https://github.com/alphagov/govuk-design-system-backlog/issues/43#issuecomment-1160139594)
Iemand van GOV.UK legt uit wat de redenen zijn waarom zij geen gebruik maken van de native Date input en Date picker.

[Maybe You Don’t Need a Date Picker - Adrian Roselli](https://adrianroselli.com/2019/07/maybe-you-dont-need-a-date-picker.html)
Artikel waarin Adrian Roselli beschrijft dat de native implementatie van `input type ="date"` niet toegankelijk is. Daarbij doet hij ook een voorstel op basis van `input type ="text"`.

[Date and Time Pickers for All - Adobe Spectrum](https://react-spectrum.adobe.com/blog/date-and-time-pickers-for-all.html)
Artikel waarin wordt aangegeven dat Adobe een Date Picker heeft ontwikkeld met de nadruk op internationalisering, toegankelijkheid, flexibiliteit en aanpasbaarheid. 

[Is input type=”date” ready for use in accessible websites? - Hassell Inclusion](https://www.hassellinclusion.com/blog/input-type-date-ready-for-use/)
Artikel waarin wordt beschreven hoe input type="date" in HTML5, bedoeld om het invoeren van datums te vereenvoudigen en gebruikersfouten te verminderen, wordt ondersteund door browsers en ondersteunende technologieën, en hoe het de toegankelijkheid van formulieren beïnvloedt.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-date-input--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/date-input)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29185&t=YiutWcy7egBq6beS-11)

---

### Screenshots / Varianten

**Storybook**

Default
Standaard date input die gebruikers helpt om een datum in te voeren.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/e0ab0b13-fadb-4813-ad2b-bc64460c7873">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/036f9ada-96c1-470c-b208-3d95058affa8">

Invalid
Date input waarbij een foutmelding wordt getoond bij ongeldige invoer.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/7580e185-4a70-40ef-98a7-10ac46fc62c5">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/642e4bb0-9116-46b2-9e74-c499fc463df6">

Disabled
Date input waarbij geen interactie mogelijk is. 
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/6e890285-78d5-4d12-8890-0d8a44e7bbf8">

Small container
<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/85b64698-c59f-44fb-a9c6-fcaed1ef414c">


**Figma**

Date input met de states default, hover, focus, error en disabled. 
De date picker die te zien is, is enkel een voorbeeld. In de implementaties worden browser/OS default datepickers gebruikt voor een verhoogd gebruiksgemak en onderhoudsgemak.
<img width="1115" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/301ac04b-e50e-438c-8854-1b91fe649887">

---

### Notities
- Input type ="date". 
- Native Calendar aanwezig.
- Calendar wordt geactiveerd wanneer je de icon klikt.
- Date format: dd-mm-jjjj. 
- In design wordt de stijl van de Date input wel overgenomen, maar voor de Date picker wordt enkel de juiste interactiekleur gebruikt.
- Input heeft een breedte die logisch is afgestemd op de verwachte invoer.
- Het is mogelijk om de input te wissen en de datum van vandaag aan te klikken. 

Interactie
- Je kunt een datum invullen door op 'dd', 'mm' en 'jjjj' te klikken en vervolgens een getal in te voeren. Het is niet mogelijk om vrijuit een datum in te vullen. 
- Als je een getal zoals 4 invult bij bijv. 'dd', dan wordt de meest rechter invoer als eerst ingevuld (dus: '04').
- Wanneer je een getal hebt ingevuld bij bijv. 'dd', dan word de focus automatisch gelegd op 'mm'. 
- Bij 'jjjj' is het mogelijk om meer dan 4 karakters in te voeren. De Date input wordt invalid. 
- Met backspace verwijder je de invoer voor 'dd', 'mm' en 'jjjj'. Je kunt niet de gehele invoer in één keer verwijderen.

**Comment by Rozerinay** [comment]
## DSO Toolkit

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/form/date-picker)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/core-date-picker--default)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/date-picker)

---

### Screenshots / Varianten

**Storybook**

Default
Standaard date input die gebruikers helpt om een datum in te voeren.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/3778b74a-8f07-49f3-ad4e-72cff8bd64d7">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/51035f0a-f7ed-4de9-a880-4f0bd1866a42">

Disabled
Date input waarbij geen interactie mogelijk is.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/61abf4ef-a80e-466f-b3a1-588d6d0848f8">

Invalid
Date input waarbij een foutmelding wordt getoond bij ongeldige invoer.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c1e29c46-b4d7-4dc0-8846-55cbae23a3e3">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/fce93e62-5594-40b6-a57c-18ddadd36369">

With value
Date input waarbij een datum als waarde is ingevuld.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/1d242c6f-d986-4909-840d-7c523312703a">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d4914b52-6538-47ac-9f7b-12ea67fabbea">

With min and max
Date input waarbij de maand en het jaar zijn vastgesteld en alleen kan worden gekozen voor een dag.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/0a78cdfe-fad1-4bc2-b5a6-556c2dbead06">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d65dda0f-15d0-4b43-9cb4-b8245212e2a5">

Month range
Date input waarbij een begin- en einddatum van een maand zijn vastgesteld om daarbinnen te kiezen.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/92e387dd-a0d7-42ee-aecf-9f3f63c58164">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/5a1a454f-4229-4bf7-84e8-045ee9c64eee">

With label
Date input met een label. 
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c86ee161-7390-46fd-bc2c-e2973b74bf09">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/4f2d4b98-7bf1-4ccc-ad86-2035a2e5be70">

Narrow input
Date input waarbij de breedte van het veld is afgestemd op de invoer.
<img width="200" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/72071080-47e8-44e0-84f0-4890d7b0fcf4">

<img width="270" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/04cda4fb-31fe-4e8a-ad31-9fbfbfc70b21">

---

### Notities
- Heet 'Date picker'.
- Input type="date".
- Native Calendar aanwezig.
- Calendar wordt geactiveerd wanneer je de icon klikt.
- Date format: dd-mm-jjjj.
- Input heeft een breedte die logisch is afgestemd op de verwachte invoer (narrow input) of de standaardbreedte behoudt (default).
- Bij de Disabled state is geen icon aanwezig.
- Het knopje om de modal voor het kiezen van een datum te openen heeft het aria-expanded attribuut en als label "kies datum", daaraan wordt "geselecteerde datum [datum]"' toegevoegd als al een datum is geselecteerd. Dit is bedoeld als extra hint voor de gebruiker.
- Heeft dezelfde interactie als bij Amsterdam (waar de interactie staat uitgeschreven).

**Comment by Rozerinay** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/datepicker)

---

### Screenshots / Varianten

**Datepicker**
De standaard datepicker wordt gebruikt voor het selecteren van één specifieke datum in de nabije toekomst of het recente verleden.

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/006fbbdb-87d0-438e-a268-bf419fcf2a71">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/74c40360-6886-4bcc-acb6-d37057c7d4d4">

**Datetimepicker**
Gebruik de datetimepicker wanneer je de gebruiker vraagt een specifieke datum en tijd in te voeren.

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/67e91a25-c6fc-47b5-8877-56ec638092e8">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/cf74420b-aa97-42a9-b255-b9593d17574a">

**Timepicker**
Gebruik de timepicker wanneer je de gebruiker vraagt een specifieke tijd in te voeren.
(Deze wordt niet in elke browser ondersteund)

<img width="200" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/79584116-5ef0-4986-8b87-5c71bac14242">

<img width="200" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/e23d73ea-0b3b-4805-aecb-6cfe4487bc68">

**States**

Uitgeschakeld veld
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d82b0c3c-3238-4480-b9fd-9f05ba01090e">

Veld met focus
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c5ee4e50-d95b-46c6-955a-2e45410fc8d2">

Verkeerde invoer
Wanneer een gebruiker een verplichte vraag verkeerd of niet ingevuld heeft, tonen we een foutmelding.
<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/df14e404-5096-4086-b585-ff8260e5eb09">

---

### Notities
- Heet 'Datepicker'.
- Datepicker heeft input type="date", Datetimepicker "datetime-local" en Timepicker "time".
- Native Calendar en Timepicker aanwezig.
- Calendar en Timepicker worden geactiveerd wanneer je op de button klikt.
- Date format: dd-mm-jjjj.
- Time format: --:--
- Input heeft een breedte die logisch is afgestemd op de verwachte invoer.
- Heeft dezelfde interactie als bij Amsterdam (waar de interactie staat uitgeschreven).
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## OpenGemeenten

[Pattern Lab](https://patternlab.lenteveld.nl/?p=molecules-Date)

---

### Screenshots / Varianten

**Date**
<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/4226d8fa-dc95-4109-8be2-3492cc2f5d97">

<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/ce416f9e-bd6c-48f7-8c7a-cb4af4b20825">


**Date description**
<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/51425537-8fef-4387-87f8-17a02aa95702">

**Date error**
<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/cd84d4b6-dae6-4d03-96db-0782202f03b5">

**Date required**
<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/905bbcb5-b32e-47f1-a282-359177ee7b9c">

**Date value**
<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c176c441-590d-421f-b062-790ce8654192">

---

### Notities
- Heet 'Date'.
- Input type="date".
- Native Calendar aanwezig.
- Calendar wordt geactiveerd wanneer je op de icon klikt.
- Date format: dd-mm-jjjj.
- Input heeft een standaardbreedte.
- Heeft dezelfde interactie als bij Amsterdam (waar de interactie staat uitgeschreven).

**Comment by Rozerinay** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-date-input--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-field-dateinput)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=258-7487&t=7Gkke4pfh1zgidQo-11)

---

### Screenshots / Varianten

**Storybook**
Default
<img width="200" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/122e3d8c-1558-478e-88ee-b9268debf590">

<img width="300" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/0c1457b4-b7d2-4b51-b8c1-3f4afa5f1be7">

**Figma**
Date input met de states default, disabled en invalid.
<img width="400" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c3e66ae4-f718-4d2c-89b9-c54490e284a2">

---

### Notities
- Heet 'Date input field' op de documentatiewebsite en 'Date input' in Storybook'.
- Input type="date".
- Native Calendar aanwezig.
- Calendar wordt geactiveerd wanneer je op de icon klikt.
- Date format: dd-mm-jjjj.
- Heeft de mogelijkheid om de attributen value, min, max, step en size in te stellen. 
- Input heeft een breedte die logisch is afgestemd op de verwachte invoer.
- Heeft dezelfde interactie als bij Amsterdam (waar de interactie staat uitgeschreven).

**Comment by sergei-maertens** [comment]
Open Formulieren (https://open-formulieren.github.io/open-forms-sdk/?path=/docs/pure-react-components-forms-datefield--docs) heeft twee varianten hiervan, waarvan ik de tweede niet zie, maar komt wel overeen met het GOV.uk artikel. Voor "bekende" datums zoals geboortedatums maken we gebruik van een "input group" zonder date picker, die laatste zit daar namelijk behoorlijk in de weg. Niet zo'n probleem met native `type="data"` inputs omdat je daar wel nog kan typen. Eén belangrijk verschil daar is dat de volgende van de elementen op basis van de locale verandert - US locale doet bijvoorbeeld `mm-dd-yyyy`

De custom datepicker wordt ook gebruikt om calendar-ranges/dagen uit te kunnen blocken, dat kan met native inputs niet.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Date Input op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=date+input+%23188)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Date Input op het Candidate bord](https://github.com/orgs/nl-design-system/projects/80/views/1?filterQuery=Date+Input)
